core/modules/node/node.module

```diff
@@ -3051,6 +3059,47 @@ function _node_query_node_access_alter($query, $type) {
 
 }
 
+/**
+ * Exclude nodes from the search query where the node has a hidden path
+ * and the current user doesn't have permission to bypass the restriction.
+ */
+function _node_search_query_alter(QueryAlterableInterface $query) {
+  global $user;
+
+  if ($user->uid == 1) {
+    return;
+  }
```

Comment:
This IF statement is unnecessary since `user_access()` below will already do the same check.